.github/workflows/build.yaml

```diff
@@ -0,0 +1,58 @@
+name: Build ADI artifacts
+
+on: [pull_request]
+
+# The current Buildroot container installs GCC15, while install-compilers.sh
+# downloads and uses GCC13 for building host tools. Mixing GCC13-compiled C++
+# host tools with the system libstdc++ from GCC15 leads to ABI mismatches.
+# To avoid this, we explicitly set CC/CXX to gcc-15/g++-15 so the build 
+# consistently uses the GCC15 toolchain.
+env:
+  CC: gcc-15
+  CXX: g++-15
+
+jobs:
+  build:
+    runs-on: self-hosted
+
+    strategy:
+      fail-fast: false
+      matrix:
+        defconfig:
+          - adi_sc598_ezkit_defconfig
+
+    steps:
+      - name: Checkout br2-external
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Cache download directory
+        uses: actions/cache@v4
+        with:
+          path: ~/.buildroot-dl
+          key: ${{ runner.os }}-buildroot-dl-${{ matrix.defconfig }}-${{ hashFiles('Makefile') }}
```

Comment:
This hash isn't being computed. I assume this would look for `br2_external/Makefile`. I think for now we don't need to hash any files. The build won't cache new changes so in theory the build time might slowly increase, but right now that is less of a concern compared to the cost of doing a fully clean build.